test/e2e/support/custom.ts

```diff
@@ -862,75 +862,54 @@ Cypress.Commands.add('getCurrentViewportSize', function () {
   }));
 });
 
-Cypress.Commands.add('showNavGroups', (open) => {
-  cy.get('body').then((body) => {
-    const isVisible = !!body.find('[data-testid=page-navigation]').length;
-    const isDialogOpen = !!body.find('[data-testid=page-navigation-dialog]').length;
+Cypress.Commands.add('showNavGroups', () => {
+  cy.findByRole('button', { name: 'Skjemasider' }).click();
+  cy.findByRole('dialog', { name: 'Skjemasider' }).should('be.visible');
+  cy.findByRole('dialog', { name: 'Skjemasider' }).should('have.css', 'opacity', '1');
+});
 
-    if (open && isVisible) {
-      return;
-    }
-    if (open && !isVisible) {
-      cy.findByRole('button', { name: 'Skjemasider' }).click();
-      cy.findByRole('dialog', { name: 'Skjemasider' }).should('be.visible');
-      cy.findByRole('dialog', { name: 'Skjemasider' }).should('have.css', 'opacity', '1');
-    }
-    if (!open && isDialogOpen) {
-      cy.findByRole('dialog', { name: 'Skjemasider' }).within(() => cy.findByRole('button', { name: 'Lukk' }).click());
-      cy.findByRole('dialog', { name: 'Skjemasider' }).should('not.exist');
-    }
-  });
+Cypress.Commands.add('hideNavGroups', () => {
+  cy.findByRole('dialog', { name: 'Skjemasider' }).within(() => cy.findByRole('button', { name: 'Lukk' }).click());
+  cy.findByRole('dialog', { name: 'Skjemasider' }).should('not.exist');
 });
 
 Cypress.Commands.add('navGroup', (groupName, pageName) => {
   if (pageName) {
-    cy.get('[data-testid=page-navigation]').then((container) =>
-      cy
-        .findByRole('button', { name: groupName, container })
-        .parent()
-        .then((container) => cy.findByRole('button', { name: pageName, container })),
-    );
+    cy.get('[data-testid=page-navigation]')
+      .findByRole('button', { name: groupName })
+      .parent()
+      .findByRole('button', { name: pageName });
```

Comment:
`findByRole` does not get scoped by the subject its called on, so in this case only the last `.findByRole` really matters (except for asserting that a group with groupName also exists, but it is not necessarily the "parent" of pageName). If there are multiple pages with the same display name but in different groups this will not be able to distinguish the two. It will not distinguish between the navigation menu and other buttons with the same name either

Reply:
Hmm, I'll revert the changes I made to this function. To me it just seemed like wrapping in `.then()` was not really need (and I suspected it broke automatic retries in Cypress), but having run the test 5x successfully now in the problematic branch, I guess this was not really the problem. :pray: